File: chrome-extension/src/types/dictionary.ts
export type Example = {
  text: string;
  translation: string;
};

export type Meaning = {
  text: string;
  translation: string;
  examples: Example[];
};

export type Definition = {
  pos: string;
  pronunciationLink: string;
  meaning: Meaning;
};

export type DictionaryPage = {
  symbol: string;
  definitions: Definition[];
};

File: services/src/domain/dictionary.go
package domain

type Example struct {
	Text        string `json:"text"`
	Translation string `json:"translation"`
}

type Meaning struct {
	Text        string    `json:"text"`
	Translation string    `json:"translation"`
	Examples    []Example `json:"examples"`
}

type Definition struct {
	POS               string  `json:"pos"`
	PronunciationLink string  `json:"pronunciationLink"`
	Meaning           Meaning `json:"meaning"`
}

type DictionaryPage struct {
	Symbol      string       `json:"symbol"`
	Definitions []Definition `json:"definitions"`
}
